src/GeniusYield/Providers/Node/AwaitTx.hs

```diff
@@ -0,0 +1,75 @@
+{-|
+Module      : GeniusYield.Providers.Node.AwaitTx
+Description : AwaitTx provider using local node. Inefficient, for testing purposes only.
+Copyright   : (c) 2023 GYELD GMBH
+License     : Apache 2.0
+Maintainer  : support@geniusyield.co
+Stability   : develop
+
+**INTERNAL MODULE**
+-}
+
+module GeniusYield.Providers.Node.AwaitTx (
+    nodeAwaitTxConfirmed
+) where
+
+import           Control.Concurrent               (threadDelay)
+import           Control.Exception                (throwIO)
+import           Control.Monad                    (unless)
+
+import qualified Cardano.Api                      as Api
+
+import           GeniusYield.Providers.Node.Query
+import           GeniusYield.Types
+
+{- TODO: Perhaps it's possible to do this more efficiently using one of the node mini protocols.
+
+Idea using chain sync:-
+
+On each RollForward event, store the transaction ids within that block, and the block number
+with the transactions. The number of confirmations for each of those transactions is their block
+number subtracted from the current block.
+
+Since ChainTip doesn't return current block number, and instead returns the current slot number - maybe
+store block's beginning slot number (absolute) instead (available in Rollforward event).
+If (chain tip slot - tx block slot) >= 3 * k / active slot coeff, then at least k blocks have been
+created since the tx - thus, there have been at least k confirmations.
+
+See: https://docs.cardano.org/about-cardano/learn/chain-confirmation-versus-transaction-confirmation/
+-}
+nodeAwaitTxConfirmed :: GYEra -> Api.LocalNodeConnectInfo -> GYAwaitTx
+nodeAwaitTxConfirmed era info p@GYAwaitTxParameters{..} txId = go 0
+  where
+    go attempt
+      | attempt >= maxAttempts = throwIO $ GYAwaitTxException p
+      | otherwise = do
+          utxos <- nodeUtxosFromTx era info txId
```

Comment:
True, but very unlikely for privnet testing I'd think. This is an internal module only meant for privnet anyway.

But I will try and see if there's a way to avoid it. At the very least, I will put a comment acknowledging this issue.

Reply:
Yes, let's revisit it at a later point.